rig-core/src/providers/aliyun/embedding.rs

```diff
@@ -0,0 +1,239 @@
+
+        let response = self
+            .client
+            .post("/compatible-mode/v1/embeddings")
+            .json(&request)
+            .send()
+            .await?
+            .error_for_status()?
+            .json::<ApiResponse<aliyun_api_types::EmbeddingResponse>>()
+            .await?;
+
+        match response {
+            ApiResponse::Ok(response) => {
+                let docs = documents
+                    .into_iter()
+                    .zip(response.data)
+                    .map(|(document, embedding)| embeddings::Embedding {
+                        document,
+                        vec: embedding.embedding,
+                    })
+                    .collect();
+
+                Ok(docs)
+            }
+            ApiResponse::Err(err) => Err(EmbeddingError::ProviderError(err.message)),
+        }
+    }
+}
+
+// =================================================================
+// Aliyun API Types
+// =================================================================
+/// Type definitions for Aliyun Embedding API responses
+/// Follows OpenAI-compatible API structure
+#[allow(dead_code)]
+mod aliyun_api_types {
+    use serde::Deserialize;
+
+    /// Response structure for embedding requests
+    #[derive(Debug, Deserialize)]
+    pub struct EmbeddingResponse {
+        pub data: Vec<EmbeddingData>,
+        pub model: String,
+        pub object: String,
+        pub usage: Usage,
+        pub id: String,
+    }
+
+    /// Individual embedding data for a single input document
+    #[derive(Debug, Deserialize)]
+    pub struct EmbeddingData {
+        pub embedding: Vec<f64>,
+        pub index: usize,
+        pub object: String,
+    }
+
+    /// Token usage statistics for the embedding request
+    #[derive(Debug, Deserialize)]
+    pub struct Usage {
+        pub prompt_tokens: usize,
+        pub total_tokens: usize,
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crate::embeddings::embedding::EmbeddingModel as EmbeddingModelTrait;
+
+    #[tokio::test]
+    async fn test_embed_texts() {
+        let client = Client::from_env();
+        let model = EmbeddingModel::new(client, EMBEDDING_V1, None);
+
+        // Test embedding for a single document
+        let documents = vec!["Hello, world!".to_string()];
+        let embeddings = model.embed_texts(documents).await.unwrap();
+
+        assert_eq!(embeddings.len(), 1);
+        assert_eq!(embeddings[0].vec.len(), 1536);
+
+        // Test embedding for multiple documents
+        let documents = vec!["Hello, world!".to_string(), "This is a test".to_string()];
+        let embeddings = model.embed_texts(documents).await.unwrap();
+
+        assert_eq!(embeddings.len(), 2);
+        assert_eq!(embeddings[0].vec.len(), 1536);
+        assert_eq!(embeddings[0].document, "Hello, world!");
+        assert_eq!(embeddings[1].vec.len(), 1536);
+        assert_eq!(embeddings[1].document, "This is a test");
+    }
```

Comment:
For testing, it's important that we don't use module testing that directly relies on API keys and charges as it would be quite costly in non-related PRs. We are working on better automated integration testing for releases but for now, the testing in this module should only be testing data conversions and/or using mocking.

Please see other embedding providers for some reference on testing.